app/models/user.rb

```diff
@@ -0,0 +1,38 @@
+require 'bcrypt'
+require 'data_mapper'
+
+module TheApp
+  module Models
+    class User
+      include DataMapper::Resource
+
+      attr_reader :password
+      attr_accessor :password_confirmation
+
+      property :id, Serial
+      property :name, String
+      property :email, String, required: true
+      property :password_digest, Text
+      property :username, String, required: true
+
+      validates_confirmation_of :password
```

Comment:
Prefer the new style validations `validates :column, confirmation: value` over `validates_confirmation_of`.